BUG: fix contrib.gen_coeffs Isat2 temperature dependence

mikofski opened the pull request:
* in residual, assume "isat2" in x is at STC, ie "isat2_t0" and calculate
isat2 using `diode.isat_t()`
* also fix all documentation in gen_coeffs to render okay
* make example a callable script, allow user to specify whether they want
to fit just STC or all IEC61853, show generated coeffs, and save figure
* futurize Bennet's module mismatch simulator for py36
* add contrib, gen_coeffs, and module mismatch simulator to docs in new
api section called contrib
* add example output figures
* fix intersphinx, must use https, leave off trailing slash
* don't ignore git conflict files, so they can be removed without hiding
* ignore the new .pytest_cache folder, move python3 __pycache__ folder to
#build
* fix underline in README.rst, should be as long as the title

mikofski commented:
@chetan201 somehow we missed this 7 months ago, we made the change to PVMismatch to use temperature in Isat2, but not in the coefficient generation

I also took this opportunity to add `gen_coeffs` to the documentation.

One result of this, is that it highlights the need for PVMismatch to account for irradiance effect on shunt resistance or some other algorithm.

mikofski commented:
here you can see without Rsh irradiance correction the errors are small near 1 sun, but increase as irradiance decreases:

![stc](https://user-images.githubusercontent.com/1385621/45910488-658d6000-bdbe-11e8-90dd-4c6e614b1f6d.png)

But if I change make `Rsh = Rsh_E0 / Ee` so it's dependent on irradiance then the errors at low light are around the same as at 1 sun

![stc](https://user-images.githubusercontent.com/1385621/45910517-b604bd80-bdbe-11e8-90f8-df27e30fc907.png)

chetan201 commented on pvmismatch/contrib/gen_coeffs/__init__.py:
```diff
@@ -11,25 +11,37 @@
 
-    :param pvmodule: PV module to be tested
-    :type pvmodule: dict
+    :param dict pvmodule: PV module to be tested
```
Thanks for adding docstrings everywhere! 
Our future users will thanks us plenty for that :)

chetan201 commented:
@mikofski 
When I run this locally ```example.py``` I get this - The RMSE is quite higher than what your results show. [This is what I am running.](https://github.com/SunPower/PVMismatch/tree/cc_issue89)
<img width="717" alt="image" src="https://user-images.githubusercontent.com/19290930/45932133-bfac3380-bf2c-11e8-8650-94fe5cb32fa1.png">

mikofski commented:
Sorry, I had changed the calculation of the rms error relative to the STC conditions after I made the original plots above, see 257e5980f70585e7f1d40950955bb1808a734821  and I've changed them again in 74a0952bf01d8ed11a6b8bb710715e43ed17edba to use the actual max power point (`mpp`) instead of the corresponding voltage corresponding to the IEC61853 Imp, and the current corresponding to the IEC61853 Vmp, which was introducing extra error, and wasn't a good comparison. Also note this is not the residual of the solver, it's the RMS error between the four points on the IV curve for all irradiances in the plot. I also output the Isc and Voc on the plots next to the irradiance level to check that it's calculated correctly, in case that was introducing error too. Here's the RMSE equations used

    rmse = sqrt(sum(
        (Vmp_calc - Vmp_meas/Ns)**2 / (Vmp0/Ns)**2,        
        (Imp_calc - Imp_meas/Np)**2 / (Imp0/Np)**2,        
        (Voc_calc - Voc_meas/Ns)**2 / (Voc0/Ns)**2,        
        (Isc_calc - Isc_meas/Np)**2 / (Isc0/Np)**2,        
    ) / 7 / 4 ) * 100  # 7 irradiance measures, 4 points, 100%

    python pvmismatch\contrib\gen_coeffs\example.py STC examples

![stc](https://user-images.githubusercontent.com/1385621/45938540-c3b57100-bf7f-11e8-8c56-f15e5d791186.png)

    python pvmismatch\contrib\gen_coeffs\example.py IEC61853 examples

![iec61853](https://user-images.githubusercontent.com/1385621/45938542-c9ab5200-bf7f-11e8-91ee-d561136ab5c6.png)

This shows there's more spread in the errors from fits using only STC data vs. the entire IEC61853 data set. I think this error is mostly at Voc, you can see that the curves, don't cross through the IEC measurements. This might be because of #91 or #90 or maybe even because there is still a missing temperature band gap energy dependence. I think the power error relative to STC is probably more useful. This is also shown on the plot, and I calculated it like this:

    dellta_stc = (Pmp_calc - Pmp_meas) / Pmp0 * 100

Note, the reason I used STC everywhere in the denominator was so that at very small irradiance, the errors are not exaggerated because the number are so small, eg: the power at 0.1[sun] is 11X less than at 1.1[suns] so should the relative error would have been (`deltaP / a very small number`) which would have exaggerated the error compared to 1[sun]. OK?

chetan201 commented:
@mikofski 
I verified the the new plots and they look good. Thanks for improving the gen_coeffs module further.
